hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/InternalGenericRecord.java

```diff
@@ -65,6 +65,13 @@
  */
 public interface InternalGenericRecord extends GenericRecord {
 
+    /**
+     * @param fieldName the name of the field
+     * @return true if field exists in the schema/class definition. Note that
+     * returns true even if the field is null.
+     */
+    boolean hasField(@Nonnull String fieldName);
```

Comment:
Did you forget to delete `hasField ` method here?

Reply:
No, this is the InternalGenericRecord interface, where we use this method during queries. So, we need it for our own implementations, but the user cannot use it, as it is not in the public GenericRecord interface

Reply:
I see, you need it for Portable as well.